src/Microsoft.Extensions.Configuration/DefaultConfigurationBuilderExtensions.cs

```diff
@@ -21,7 +21,7 @@ public static IConfigurationBuilder Include(this IConfigurationBuilder configura
             {
                 throw new ArgumentNullException(nameof(configurationBuilder));
             }
-            configurationBuilder.Add(new IncludedConfigurationProvider(includedConfiguration));
+            configurationBuilder.Add(new IncludedConfigurationSource { Source = includedConfiguration });
```

Comment:
It was a cool feature, but we didn't think the current implementation was ideal (remember we have a bug to make avoid copying the state?). Also we would need to wire up change notifications to make it complete.

So if it is confirmed that nobody is using it, I agree removing it is the best option.

cc @davidfowl @DamianEdwards who originally insisted it was necessary, just in case.